### PR TITLE
fix: member joined system message is missing after joining conversati…

### PR DESCRIPTION
**Pull request by MohamadJaara**
…on with a guest link

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

self invoked member joined system messages were hidden to avoid displaying something like `You added You to the conversation`

### Solutions

map self invoked member joined to the correct message type 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1564/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1564/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/SystemMessageItem.kt [hunk 1]
```diff
@@ -158,6 +158,7 @@ private fun getColorFilter(message: SystemMessage): ColorFilter? {
         is SystemMessage.MissedCall.YouCalled -> null
         is SystemMessage.Knock -> ColorFilter.tint(colorsScheme().primary)
         is SystemMessage.MemberAdded -> ColorFilter.tint(colorsScheme().onBackground)
+        is SystemMessage.MemberJoined -> ColorFilter.tint(colorsScheme().onBackground)
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/SystemMessageItem.kt:161]: `MemberAdded` or `MemberJoined`, not the same ?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -928,4 +928,6 @@
     <string name="new_device_dialog_other_user_title">Your account “%1$s (@%2$s)” was used on:</string>
     <string name="new_device_dialog_other_user_message">%1$s\nFrom: %2$s \n\nIf you didn’t do this, remove the device (navigate to “Manage Devices” in the Settings section of this account), and reset your password.</string>
     <string name="new_device_dialog_other_user_btn">Switch Account</string>
+    <string name="label_system_message_joined_the_conversation_by_self">%1$s joined the conversation</string>
+    <string name="label_system_message_joined_the_conversation_by_other">%1$s joined the conversation</string>
```
Comment (ohassine) [inline app/src/main/res/values/strings.xml:932]: I think we already have this string in our strings file, `last_message_joined_conversation` can't we use it ?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/SystemMessageItem.kt (hunk 1 above)
Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/SystemMessageItem.kt:161]: Added is `X added Y to the conversation`Joined is `X joined the conversation`

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (MohamadJaara) [reply app/src/main/res/values/strings.xml:932]: I did not notice that I will remove the new strings and use the old one

Comment (MohamadJaara) [reply app/src/main/res/values/strings.xml:932]: i just double checked and they are not the same strings so we have to use those

**Review by ohassine — APPROVED** [review]
LGTM 👌

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1564/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6044173)(6044173)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6044173)(6044173)-staging-compat.apk)

**Review by saleniuk — APPROVED** [review]

**Pull request merged**